### ISSUE TITLE
How to load pretrain model?

### ISSUE THREAD

**Issue by CheungBH** [open-mmlab/mmdetection#7312]
Is it done by changing the "checkpoint" here to the path of model?
https://github.com/open-mmlab/mmdetection/blob/1872bc44d04601045110186ad566c7117f9a0b2a/configs/_base_/models/faster_rcnn_r50_fpn.py#L13

**Comment by Soodep** [comment]
This is the backbone network checkpoint location. In order to load the pretrained model you can directly download the pretrained model from: https://github.com/open-mmlab/mmdetection/tree/master/configs/faster_rcnn , this is for faster_rcnn. You can find pretrained model for other methods as well. After you download the pretrained model, you can provide the location of the file when you test. For e.g.
python tools/test.py configs/pascal_voc/faster_rcnn_r50_fpn_1x_voc0712.py location_to_your_checkpointfile  --show

**Comment by CheungBH** [comment]
> This is the backbone network checkpoint location. In order to load the pretrained model you can directly download the pretrained model from: https://github.com/open-mmlab/mmdetection/tree/master/configs/faster_rcnn , this is for faster_rcnn. You can find pretrained model for other methods as well. After you download the pretrained model, you can provide the location of the file when you test. For e.g. python tools/test.py configs/pascal_voc/faster_rcnn_r50_fpn_1x_voc0712.py location_to_your_checkpointfile --show

Sorry that I haven't expressed it clearly. I don't need the normally-trained pretrained model provided by you. Instead, I trained my own model _**M1**_ with some additional layers, and I am going to train another model _**M2**_ on the basis of _**M1**_. 
In this situation, are there any interface to load the weight of _**M1**_?

**Comment by CheungBH** [comment]
I noticed that loading pretrained model is implemented by overriding the function "init_weights". But I found that the base function of "init_weights" is implemented by mmcv and no parameters can be passed in it.
Also, I found that when training renset50, the weight from **'torchvision://resnet50'** will be automatically downloaded and loaded. However, such codes are not visible, and I don't know how to achieve it when writing another backbone loading resnet50 weights.

In short, the problem is:
1. How to pass a path into "init_weights"?
2. Where are the codes achieving downloading and loading the weight from torchvision?

**Comment by RangiLyu** [comment]
> 1. How to pass a path into "init_weights"?
just write your checkpoint path to the `checkpoint`
```
        init_cfg=dict(
            type='Pretrained',
            checkpoint='YOUR_CHECKPOINT_PATH')
```

> 2. Where are the codes achieving downloading and loading the weight from torchvision?
https://github.com/open-mmlab/mmcv/blob/f527e43c1a1e52e5903410255aea1f71f005a161/mmcv/cnn/utils/weight_init.py#L481
https://github.com/open-mmlab/mmcv/blob/f527e43c1a1e52e5903410255aea1f71f005a161/mmcv/runner/checkpoint.py#L207

**Comment by huichen98** [comment]
thanksthanks
